Srinter 749

soda736 opened the pull request:
Added in display suite module to provide extra options for displaying content. I.e changing field labels.

soda736 committed:
Added display suite for extra options on dispaying content

soda736 committed:
Merge branch 'development' into SRINTER-749

Getting up to date with latest development code.

soda736 committed:
Removing remnants of toc_js field I was playing about with

ooneill547 commented on config/sync/core.extension.yml:
```diff
@@ -37,7 +37,6 @@ module:
   node: 0
   options: 0
   origins_common: 0
-  origins_layouts: 0
```
Why was this uninstalled?

soda736 replied:
I was trying to disable layout builder to see if it was conflicting with the display suite module. Forgot to turn it back on. Enabled again

soda736 committed:
Re-installing origins layouts module

soda736 committed:
Added a no results summary to the publication search view

ooneill547 reviewed (commented):
It looks good to me

omahm commented on composer.json:
```diff
@@ -23,6 +23,7 @@
         "drupal-composer/drupal-scaffold": "^2.5",
         "drupal/console": "^1.0.2",
         "drupal/core": "^8.7.0",
+        "drupal/ds": "^3.4",
```
DS is overkill for changing a label.

johangant reviewed (changes requested):
Thanks for this, @soda736: some changes I think are needed (summarising our earlier discussion):

1. You'll want to explore a preprocess hook such as preprocess_field or preprocess_node to adapt the publish date field label on the search results page.
2. You should probably look at an alternative view mode for the search results page rather than teaser, which would normally be found on a card display or listing display. I'd suggest a Search result or 'Search result with snippet' view mode as it's coming from Solr: you can create a view mode for that if you haven't already got one.
3. You don't want the config changes for Page content type: that was in there by mistake.

Just shout if you need a hand with any of these points and someone in the team will be happy to help.

soda736 committed:
Changed a label and added logic for publication search page count

soda736 committed:
Removed ds module and added devel to composer --dev

soda736 committed:
Added a new view mode for publication search results and removed ds module

soda736 committed:
Resetting basic page default view

soda736 committed:
Adding links back into basic page default view

soda736 committed:
Removing links back into content

soda736 commented:
I have made the required changes:

- I added a preprocess hook to change the field label to "Published". I also added a preprocess hook for the results count at the top of the publication search page. I couldn't apply logic in the UI for if there was only 1 result to have a singular "1 publication". 

- I have added a new view mode called "Search result with snippet" and applied this to the publication display and the publication search view.

- I made manual changes to get the basic page default page view back to its original state.

- I have also removed the display suite module along with and relating .yml files.

soda736 committed:
Changed the translatable publication label and added some comments

soda736 commented:
I have changed the translatable label to t('Published') . ':';

soda736 committed:
Adding punctuation to the end of comments for circleCI

omahm reviewed (changes requested):
Looks good. You can condense this down by half the View header message and Drupal has the formatPlural function to help. I'll show you tomorrow.

soda736 committed:
Condensing preprocess code in .theme file

johangant commented on web/themes/custom/cybersecurity/cybersecurity.theme:
```diff
@@ -26,3 +26,53 @@ function tailwindcss_preprocess_block(&$variables) {
+
+  // Check if the field label in the correct view mode and content type.
+  if ($variables['element']['#bundle'] == 'publication' && $variables['element']['#view_mode'] == 'search_result_with_snippet') {
+    $variables['label'] = t('Published') . ':';
```
Trailing space needed after the colon? Ie: `: ` rather than `:`

johangant commented on web/themes/custom/cybersecurity/cybersecurity.theme:
```diff
@@ -26,3 +26,53 @@ function tailwindcss_preprocess_block(&$variables) {
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
```
`hook_preprocess_HOOK()` should be `hook_preprocess_field()` based on the function name.

johangant commented on web/themes/custom/cybersecurity/cybersecurity.theme:
```diff
@@ -26,3 +26,53 @@ function tailwindcss_preprocess_block(&$variables) {
+}
+
+/**
+ * Implements hook_preprocess_HOOK() for views_view.
```
It's customary to use `Implements hook_preprocess_views_view()` to match your function name.

soda736 committed:
Renaming hook comments and adding extra placeholder to translatable function for circleCI test

soda736 committed:
Fixing more errors for circleCI

omahm reviewed (approved):
LGTM
